demo.py

```diff
@@ -306,42 +281,33 @@ def visualize_results(dist_mat, city_names, hubs, legs, city_lats, city_longs, c
     # G = build_graph(passenger_demand, city_names)
     # draw_graph(G, city_names, city_lats, city_longs)
 
-    dqm = build_dqm(passenger_demand, leg_cost, num_cities, p, a)
+    cqm = build_cqm(passenger_demand, leg_cost, num_cities, p, a)
 
     print("\nRunning hybrid solver...\n")
-    sampler = LeapHybridDQMSampler()
-    sampleset = sampler.sample_dqm(dqm, label='Example - DQM Airline Hubs')
+    sampler = LeapHybridCQMSampler()
+    sampleset = sampler.sample_cqm(cqm, label='Example - CQM Airline Hubs')
+    sampleset = sampleset.filter(lambda d: d.is_feasible).aggregate()
 
     print("\nInterpreting solutions...\n")
 
-    ss = list(sampleset.data(['sample']))
+    assignments = [{i: j for i in range(num_cities) for j in range(num_cities) 
+                    if sample.sample[i,j] == 1} for sample in sampleset.data()]
 
-    cost_dict = {index: get_cost(ss[index].sample, a, passenger_demand, leg_cost, num_cities) for index in range(len(ss))}
-
-    ordered_samples = dict(sorted(cost_dict.items(), key=lambda item: item[1], reverse=True))
     filenames = []
-    counter = 0
     print("\nGenerating images for output GIF...\n")
     print("\nFeasible solutions found:")
     print("---------------------------\n")
     output_string = []
-    for key, val in ordered_samples.items():
-        hubs, legs, valid = get_layout_from_sample(ss[key].sample, city_names, p)
-        if counter > 0:
-            if prev_val == val:
-                valid = False
-        if valid:
-            filenames = visualize_results(passenger_demand, city_names, hubs, legs, city_lats, city_longs, cost_dict[key], filenames, counter, verbose=False)
-            output_string.append("Hubs: "+str(hubs)+"\tCost: "+str(cost_dict[key]))
-            counter += 1
-        prev_val = val
-
-    output_string.reverse()
-    for line in output_string:
-        print(line)
+
+    for count, (sample, assignment) in enumerate(zip(sampleset.data(), assignments)):
+        hubs, legs = get_layout_from_sample(assignment, city_names, p)
+
+        filenames = visualize_results(city_names, hubs, legs, city_lats, city_longs, 
+                                        sample.energy, filenames, count, verbose=False)
+        print(f'Hubs: {hubs}\tCost: {sample.energy}')
 
     # build gif
-    print("\nSaving best solution to best_soln_found.png...\n")
+    print("\nSaving best solution to best_soln_found.png...")
     img = plt.imread(filenames[-1])
```

Comment:
Best solution is now first:
```suggestion
    img = plt.imread(filenames[0])
```